src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/Registry/CosmosDbStatusRegistryInitializer.cs

```diff
@@ -30,28 +30,33 @@ public class CosmosDbStatusRegistryInitializer : IFhirCollectionUpdater
             _queryFactory = queryFactory;
         }
 
-        public async Task ExecuteAsync(IDocumentClient client, DocumentCollection collection, Uri relativeCollectionUri)
+        public async Task ExecuteAsync(Container collection)
         {
-            EnsureArg.IsNotNull(client, nameof(client));
             EnsureArg.IsNotNull(collection, nameof(collection));
-            EnsureArg.IsNotNull(relativeCollectionUri, nameof(relativeCollectionUri));
 
             // Detect if registry has been initialized
             var query = _queryFactory.Create<dynamic>(
-                client,
+                collection,
                 new CosmosQueryContext(
-                    relativeCollectionUri,
-                    new SqlQuerySpec($"SELECT TOP 1 * FROM c where c.{KnownDocumentProperties.PartitionKey} = '{SearchParameterStatusWrapper.SearchParameterStatusPartitionKey}'")));
+                    new QueryDefinition($"SELECT TOP 1 * FROM c where c.{KnownDocumentProperties.PartitionKey} = '{SearchParameterStatusWrapper.SearchParameterStatusPartitionKey}'"),
```

Comment:
Do we need to specoty the partition in the query text as well as the QueryRequestOptions?